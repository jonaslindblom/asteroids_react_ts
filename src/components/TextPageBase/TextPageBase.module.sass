@keyframes fadeIn
    0%
        opacity: 0
    100%
        opacity: 1

.container
    background-color: #2f4f4f
    display: flex
    flex-direction: column
    align-items: center
    justify-content: center
    width: 100vw
    height: 90vh
    overflow: auto
    padding-top: 50px
    animation: fadeIn 1s linear

    .title
        margin-top: 100px
        color: tomato
        font-weight: bold
        text-align: center
        font-size: 3em
        text-shadow: 1px 1px 10px #fff, 1px 1px 10px #ccc

    .children
        display: flex
        flex-direction: column
        align-items: center
        color: tomato
        width: 85vw
        max-width: 600px
        height: 100%
        gap: 16px
        margin: 12px

    .back
        position: absolute
        top: 12px
        left: 12px

        .backArrow
            width: 48px
            height: 48px

        svg
            fill: tomato

        &:hover svg
            fill: brown
            cursor: pointer
