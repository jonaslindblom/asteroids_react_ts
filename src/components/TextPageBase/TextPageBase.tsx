import styles from "./TextPageBase.module.sass";
import { ReactComponent as BackArrow } from "./assets/back.svg";
import { Link } from "react-router-dom";

interface TextPageBaseProps {
  title: string;
  children: React.ReactNode;
}

export default function TextPageBase({ title, children }: TextPageBaseProps) {
  return (
    <div className={styles.container}>
      <div className={styles.back}>
        <Link to="/">
          <BackArrow className={styles.backArrow} data-testid="back-arrow" />
        </Link>
      </div>
      <h1 className={styles.title}>{title}</h1>
      <div className={styles.children}>{children}</div>
    </div>
  );
}
