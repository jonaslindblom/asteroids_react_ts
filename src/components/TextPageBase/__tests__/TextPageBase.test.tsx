import { render, screen, fireEvent } from "@testing-library/react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import TextPageBase from "../TextPageBase";
import { MockPage, mockPageText } from "./MockPage";

const title = "Test Title";
const children = <p>Test Children</p>;

describe("TextPageBase", () => {
  it("renders the title and children", () => {
    render(
      <MemoryRouter>
        <TextPageBase title={title}>{children}</TextPageBase>
      </MemoryRouter>
    );
    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText("Test Children")).toBeInTheDocument();
  });

  it("navigates away from the component when the link is clicked", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/text-page"]} initialIndex={0}>
        <Routes>
          <Route
            path="/text-page"
            element={<TextPageBase title={title}>{children}</TextPageBase>}
          />
          <Route path="/" element={<MockPage />} />
        </Routes>
      </MemoryRouter>
    );
    const backArrow = getByTestId("back-arrow");
    expect(backArrow).toBeInTheDocument();
    fireEvent.click(backArrow);
    expect(screen.getByText(mockPageText)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <TextPageBase title={title}>{children}</TextPageBase>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
