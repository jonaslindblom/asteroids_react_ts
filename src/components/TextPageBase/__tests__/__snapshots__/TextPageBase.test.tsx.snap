// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextPageBase > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_7eea93"
  >
    <div
      class="_back_7eea93"
    >
      <a
        href="/"
      >
        <svg
          class="_backArrow_7eea93"
          data-testid="back-arrow"
          fill="#000000"
          height="800px"
          id="Capa_1"
          viewBox="0 0 45.58 45.58"
          width="800px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g>
            <path
              d="M45.506,33.532c-1.741-7.42-7.161-17.758-23.554-19.942V7.047c0-1.364-0.826-2.593-2.087-3.113 c-1.261-0.521-2.712-0.229-3.675,0.737L1.305,19.63c-1.739,1.748-1.74,4.572-0.001,6.32L16.19,40.909 c0.961,0.966,2.415,1.258,3.676,0.737c1.261-0.521,2.087-1.75,2.087-3.113v-6.331c5.593,0.007,13.656,0.743,19.392,4.313 c0.953,0.594,2.168,0.555,3.08-0.101C45.335,35.762,45.763,34.624,45.506,33.532z"
            />
          </g>
        </svg>
      </a>
    </div>
    <h1
      class="_title_7eea93"
    >
      Test Title
    </h1>
    <div
      class="_children_7eea93"
    >
      <p>
        Test Children
      </p>
    </div>
  </div>
</DocumentFragment>
`;
